04.ls/ls.rb

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+NUMBER_OF_COL = 3
+SPACE_WIDTH = 2
+
+def get_filenames(target_path)
+  Dir.glob('*', base: target_path)
+end
+
+def output(filenames)
+  number_of_row = ((filenames.size - 1) / NUMBER_OF_COL) + 1
+  filenames_table = filenames.each_slice(number_of_row).to_a
+
+  ljust_widths = filenames_table.map { |cols| cols.map(&:size).max + SPACE_WIDTH }
```

Comment:
細かいですが、18行目に出てくるcolは数値で、ここのcolsは文字列の配列です。
通常は「colが配列になっていればcols」と考えるのが自然なので、今の命名だと誤解を招くかもしれません。
（というか、僕は一瞬「ん？」となってしまいました）

なので、こっちは

```suggestion
  ljust_widths = filenames_table.map { |filenames| filenames.map(&:size).max + SPACE_WIDTH }
```

のようにしておくといいかもしれませんね。

あと、`ljust_widths`という変数名ですが、「`ljust`用の」としてしまうと具体的すぎて変更に弱くなります。
（20行目でljustメソッドを使わなくなったら「なんでljust?」と思われてしまう）

変数名は「ほどよく抽象的な名前」が一番変更に強いです。
なので、たとえばここは「列の幅 = `col_widths`」みたいな名前がいいかもしれません。
もしくは`widths`だけでも大丈夫かなと思います。

Reply:
コメントありがとうございます。
上記２点修正したうえで次のlsコマンド2に移りたいと思います！